core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -2608,12 +2608,16 @@ class KafkaApis(val requestChannel: RequestChannel,
             addResultAndMaybeSendResponse(addPartitionsToTxnRequest.errorResponseForTransaction(transactionalId, finalError))
           }
 
+          // If the request is greater than version 4, we know the client supports transaction version 2.
+          val clientTransactionVersion = if (addPartitionsToTxnRequest.version() > 4) TransactionVersion.TV_2 else TransactionVersion.TV_0
```

Comment:
nit: should we move this to TransactionVersion?

Reply:
We can. We have the same thing for EndTxn though.